js/main.js

```diff
@@ -0,0 +1,142 @@
+
+class Card {
+    constructor(effect, mode) {
+        this.effect = effect;
+        this.mode = mode;
+        this.faceUp = false;
+    }
+    flipUp() {
+        this.faceUp = true;
+    }
+    flipDown() {
+        this.faceUp = false;
+    }
+}
+
+const table = [];
+let isComplete;
+let turn;
+let selectedCards = [];
+
+const cardSlots = document.querySelectorAll('figure');
+const selectTable = document.querySelector('main');
+const selectComplete = document.getElementById('complete')
+
+selectTable.addEventListener('click', cardFlip);
+
+function startGame() {
+    const effects = ['bold', 'italics', 'bullet', 'code', 'h1', 'h2', 'link', 'number'];
+    const modes = ['markdown', 'format'];
+    isComplete = false;
+    turn = 1;
+
+    for (const effect of effects) {
+        for (const mode of modes) {
+            table.push(new Card(effect, mode));
+        }
+    }
+
+    table.sort(() => Math.random() - 0.5)
+
+    tableSetup();
+}
+
+function tableSetup() {
+    for (let i = 0; i < table.length; i++) {
+        document.getElementById(`card${i}`).classList.add(`${table[i].effect}${table[i].mode}`, "back");
+    }
+}
+
+function cardFlip(event) {
+
+    const idx = parseInt(event.target.id.replace('card', ''));
+
+    if (table[idx].faceUp === false) {
+        table[idx].flipUp();
+        displayTable();
+        selectedCards.push(idx);
+
+        if (turn === 1) {
+            turn *= -1;
+        } else if (turn === -1) {
+            compareCards();
+            testCompletion();
+            turn *= -1;
+        }
```

Comment:
It looks like `turn` is an important variable. It controls whether the player can choose an other card, or if the game has to compare the cards selected by the player. 

Quiestion: Without adding any new variables to your game, do you see an other way of deciding if the game should let the player pick an other card or not?